External/FEXCore/Source/Interface/IR/Passes/RegisterAllocationPass.cpp

```diff
@@ -9,59 +9,173 @@
 #define SRA_DEBUG(...) // printf(__VA_ARGS__)
 
 namespace {
-  constexpr uint32_t INVALID_REG = ~0U;
-  constexpr uint64_t INVALID_REGCLASS = ~0ULL;
-  constexpr uint32_t DEFAULT_INTERFERENCE_LIST_COUNT = 128;
+  constexpr uint32_t INVALID_REG = 31;
+  constexpr uint32_t INVALID_CLASS = 7;
+  constexpr uint32_t DEFAULT_INTERFERENCE_LIST_COUNT = 120;
   constexpr uint32_t DEFAULT_NODE_COUNT = 8192;
-  constexpr uint32_t DEFAULT_VIRTUAL_REG_COUNT = 1024;
+
+  constexpr uint64_t INVALID_REGCLASS = (((uint64_t)INVALID_CLASS) << 32) | (INVALID_REG);
+
+  template<unsigned _Size = 6, typename T = uint32_t>
```

Comment:
Could do with a comment explaining why we have a bucketlist and how the constants have been chosen.

Reply:
Good point. Will get to it tomorrow

Reply:
https://github.com/FEX-Emu/FEX/issues/675

Reply:
Added the comment and removed default size of 6 (it's explicitly set in all instances anyway)